src/scripts/main.js

```diff
@@ -1,3 +1,369 @@
 'use strict';
 
-// write your code here
+const startButton = document.querySelector('.start');
+const message = document.querySelector('.message-container');
+
+const fieldSide = 4;
+const fieldRows = [...document.querySelectorAll('tr')];
+const fieldCells = [...document.querySelectorAll('td')];
+const score = document.querySelector('.game-score');
+
+function generateInEmptyCell() {
+  do {
+    const x = Math.floor(Math.random() * fieldSide);
+    const y = Math.floor(Math.random() * fieldSide);
+
+    if (fieldRows[x].children[y].textContent === '') {
+      fieldRows[x].children[y].textContent = `${Math.random() >= 0.9 ? 4 : 2}`;
+      break;
+    }
+  } while (true);
+}
+
+function addClass(cells) {
+  cells.forEach(cell => {
+    if (cell.textContent !== '') {
+      cell.className = `field-cell`;
+      cell.classList.add(`field-cell--${cell.innerText}`);
+    };
+  });
+}
+
+function calcScore(scoreElement, newCellContent) {
+  const prevScore = +scoreElement.innerText;
+  const newScore = prevScore + +newCellContent;
```

Comment:
Avoid `+ +` in one row
```suggestion
  const newScore = prevScore + Number(newCellContent);
```